cli/module_loader.rs

```diff
@@ -684,15 +684,15 @@ impl<TGraphContainer: ModuleGraphContainer>
         ..
       })) => {
         let code: ModuleCodeString = match media_type {
-          MediaType::JavaScript
-          | MediaType::Unknown
+          MediaType::Unknown
           | MediaType::Cjs
           | MediaType::Mjs
           | MediaType::Json => source.clone().into(),
           MediaType::Dts | MediaType::Dcts | MediaType::Dmts => {
             Default::default()
           }
-          MediaType::TypeScript
+          MediaType::JavaScript
```

Comment:
I don't think we should be emitting JavaScript. The fix is likely somewhere else.

Also, adding a test to demonstrate what the issue is here would be useful. See the `tests/specs` folder.

Reply:
Actually, the changes in https://github.com/denoland/deno/pull/24094/files#diff-b8f09652408b91ba94406ad379632895db8b42df251d8d8c5060f70f2e986f98 will fix this original issue.